src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -125,27 +125,20 @@ private void loadFromFile() {
 		fileChooser.addChoosableFileFilter(new FileNameExtensionFilter("*.json", "json"));
 		fileChooser.addChoosableFileFilter(new FileNameExtensionFilter("*.xml", "xml"));
 		int result = fileChooser.showOpenDialog(this);
+
 		if (result == JFileChooser.APPROVE_OPTION) {
 			File selectedFile = fileChooser.getSelectedFile();
-			String fileFormat = null;
-			fileFormat = fileChooser.getSelectedFile().getName().substring(fileChooser.getSelectedFile().getName().lastIndexOf(".") + 1);
+			String fileFormat = fileChooser.getSelectedFile().getName().substring(fileChooser.getSelectedFile().getName().lastIndexOf(".") + 1);
 			String formattedData = null;
+
 			try {
 				formattedData = new Scanner(selectedFile).useDelimiter("\\Z").next(); // 1-liner that reads whole file to string
 			} catch (FileNotFoundException e) {
 				e.printStackTrace();
 			}
-			LoadedCommand myLoadedCommand = null;
-			switch (fileFormat) {
-				case "json":
-					myLoadedCommand = new JsonCommandLoader(selectedFile.getPath(), selectedFile.getName()).loadCommandFromExternalSource(formattedData);
-					break;
-				case "xml":
-					myLoadedCommand = new XMLCommandLoader(selectedFile.getPath(), selectedFile.getName()).loadCommandFromExternalSource(formattedData);
-					break;
-				default:
-					break;
-			}
+
+			CommandLoader loader = new CommandLoaderFactory().getInstance(fileFormat, formattedData, fileChooser.getSelectedFile().getName());
```

Comment:
Try avoiding passing formattedData and file name on getting loader instance. It would make its usage and reuse easier.

Reply:
Fixed.